Detectors/TPC/qc/include/TPCQC/PID.h

```diff
@@ -62,18 +66,15 @@ class PID
   /// Dump results to a file
   void dumpToFile(std::string filename);
 
-  /// get 1D histograms
-  std::vector<TH1F>& getHistograms1D() { return mHist1D; }
-  const std::vector<TH1F>& getHistograms1D() const { return mHist1D; }
-
-  /// get 2D histograms
-  std::vector<TH2F>& getHistograms2D() { return mHist2D; }
-  const std::vector<TH2F>& getHistograms2D() const { return mHist2D; }
+  /// return a map of vectors of histograms
+  //std::map<std::string_view, std::vector<TH1*>> getMapOfHisto() { return mapHist2D; }
+  //const std::map<std::string_view, std::vector<TH1*>> getMapOfHisto() const { return mapHist2D; }
+  std::map<std::string_view, std::vector<std::unique_ptr<TH1>>> getMapOfHisto() { return mapHist2D; }
+  const std::map<std::string_view, std::vector<std::unique_ptr<TH1>>> getMapOfHisto() const { return mapHist2D; }
 
  private:
-  std::vector<TH1F> mHist1D{};
-  std::vector<TH2F> mHist2D{};
-
+  //std::map<std::string_view, std::vector<TH1*>> mapHist2D;
+  std::map<std::string_view, std::vector<std::unique_ptr<TH1>()>> mapHist2D;
```

Comment:
```suggestion
  std::unordered_map<std::string_view, std::vector<std::unique_ptr<TH1>>> mapHist2D;
```
`unordered_map` should do.